src/cmd/domain.go

```diff
@@ -0,0 +1,233 @@
+package cmd
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"github.com/creasty/defaults"
+	"github.com/opslevel/cli/common"
+	"github.com/opslevel/opslevel-go/v2023"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"os"
+	"strings"
+)
+
+var assignSystemCmd = &cobra.Command{
+	Use:        "system ID|ALIAS SYSTEM",
+	Short:      "Add a system to a domain",
+	Example:    `opslevel create system my_domain my_system`,
+	Args:       cobra.ExactArgs(2),
+	ArgAliases: []string{"DOMAIN_ID", "DOMAIN_ALIAS", "SYSTEM"},
+	Run: func(cmd *cobra.Command, args []string) {
+		key := args[0]
+		system := args[1]
+
+		domain, err := getClientGQL().GetDomain(key)
+		cobra.CheckErr(err)
+		common.WasFound(domain.Id == "", key)
+
+		addErr := domain.AssignSystem(getClientGQL(), system)
+		cobra.CheckErr(addErr)
+		fmt.Printf("add system '%s' to domain '%s'\n", system, domain.Name)
+	},
+}
+
+var createDomainCmd = &cobra.Command{
+	Use:   "domain",
+	Short: "Create a domain",
+	Long: `Create a domain
+
+cat << EOF | opslevel create domain -f -
+name: "My Domain"
+description: "Hello World Domain"
+ownerId: "Z2lkOi8vb3BzbGV2ZWwvVGVhbS83NjY"
+note: "Additional details"
+EOF
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		input, err := readDomainCreateInput()
+		cobra.CheckErr(err)
+		result, err := getClientGQL().CreateDomain(*input)
+		cobra.CheckErr(err)
+		fmt.Println(result.Id)
+	},
+}
+
+var deleteDomainCmd = &cobra.Command{
+	Use:        "domain ID|ALIAS",
+	Short:      "Delete a domain",
+	Long:       `Delete a domain`,
+	Args:       cobra.ExactArgs(1),
+	ArgAliases: []string{"ID", "ALIAS"},
+	Run: func(cmd *cobra.Command, args []string) {
+		key := args[0]
+		err := getClientGQL().DeleteDomain(key)
+		cobra.CheckErr(err)
+		fmt.Printf("deleted '%s' domain\n", key)
+	},
+}
+
+var getDomainCmd = &cobra.Command{
+	Use:        "domain ID|ALIAS",
+	Short:      "Get details about a domain",
+	Long:       `Get details about a domain`,
+	Args:       cobra.ExactArgs(1),
+	ArgAliases: []string{"ID", "ALIAS"},
+	Run: func(cmd *cobra.Command, args []string) {
+		key := args[0]
+		result, err := getClientGQL().GetDomain(key)
+		cobra.CheckErr(err)
+		common.WasFound(result.Id == "", key)
+		common.PrettyPrint(result)
+	},
+}
+
+var getDomainSystemCmd = &cobra.Command{
```

Comment:
Any reason to not merge this logic into `get` and `list` domain?  This way it can be part of the JSON output of those commands.

Reply:
I thought we were following the same pattern as groups and having separate commands to get the "child" resources.  But I can add the `ChildSystems` call in the get and list and add the systems to the domain object but we don't have a field for that so I thought we were going to leave it separate.  If we include it in the domain output would it not be better to just have a hydrate function like on services?  I thought we decided against this last time we talked about it though.